### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by stewartboogert**
Updated README to explain more clearly the capability of pyg4ometry, also a few examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,31 @@ Carlo (Geant4 and Fluka) geometries.
...
+- Converting from GDML to FLUKA and MCNP
+- Exporting mesh geometries from GDML to VTP, OBJ, VRML etc
+- Python bindings to CGAL allowing complex mesh manipulation (e.g. hole filling, remeshing)
+
+[![IMAGE ALT TEXT HERE](https://img.youtube.com/vi/OPvQFZsFvhs/0.jpg)](https://www.youtube.com/watch?v=OPvQFZsFvhs)
+
+Many people and groups have used it for various tasks
+
+- Geometries for BDSIM Geant4 simulation of accelerators
+- [Legend experiment](https://indico.cern.ch/event/1252095/contributions/5592424/attachments/2730430/4746429/202310-PyHEP.pdf)
```
Comment (ManuelHu) [inline README.md:37]:
```suggestion
- [LEGEND experiment](https://indico.cern.ch/event/1252095/contributions/5592424/attachments/2730430/4746429/202310-PyHEP.pdf)
```

**Comment by gipert** [comment]
Thanks @stewartboogert and @ManuelHu, I included a showcase code snippet -- what do you think?

Problem: it crashes:
```
Traceback (most recent call last):
  File "/home/gipert/test.py", line 30, in <module>
    viewer.addLogicalVolume(reg.getWorldVolume())
  File "/user/.local/lib/python3.10/site-packages/pyg4ometry/visualisation/VtkViewer.py", line 369, in addLogicalVolume
    self.addLogicalVolumeRecursive(logical, mtra, tra)
  File "/user/.local/lib/python3.10/site-packages/pyg4ometry/visualisation/VtkViewer.py", line 586, in addLogicalVolumeRecursive
    self.addMesh(
  File "/user/.local/lib/python3.10/site-packages/pyg4ometry/visualisation/VtkViewer.py", line 759, in addMesh
    filtername = solid_name + "_filter"
TypeError: unsupported operand type(s) for +: 'Registry' and 'str'
```
why?

**Comment by gipert** [comment]
We could also add a screenshot of the viewer window below the snippet.

**Comment by stewartboogert** [comment]
> We could also add a screenshot of the viewer window below the snippet.

So need to specify on the `registry` what the world volume is. I'll make a guard against this to warn the user that pyg4ometry does not know what the top LV is. So works and I'll make a screenshot

File: README.md [hunk 2]
```diff
@@ -15,9 +15,62 @@ Carlo (Geant4 and Fluka) geometries.
...
+reg = pg4.geant4.Registry()
+
+# build the world volume
+# NOTE: syntax in the geant4 subpackage matches the Geant4 API!
+world_s = pg4.geant4.solid.Box("WorldAir", 10, 10, 10, reg, lunit="cm")
+world_l = pg4.geant4.LogicalVolume(world_s, "G4_AIR", "WorldAir", reg)
+reg.setWorld(world_l)
+
+# import an STL file
+reader = pg4.stl.Reader(g4data["stl/utah_teapot.stl"], reg)
```
Comment (ManuelHu) [inline README.md:49]:
```suggestion
reader = pg4.stl.Reader(g4data["stl/utah_teapot.stl"], registry=reg)
```
the second (unnamed) argument is the solid name, thats why it crashed.

**Review by ManuelHu — CHANGES_REQUESTED** [review]

**Comment by ManuelHu** [comment]
@stewartboogert some lines above that, the world volume is already set to the box volume (line 46 in the readme file). So setting the world volume to the teapot is not needed/wanted?

File: README.md (hunk 2 above)
Comment (stewartboogert) [reply README.md]: Ok. Fixed.

**Comment by gipert** [comment]
>So can we not do this now. I'm just writing a GLTF web viewer. So pyg4ometry can target webpages for the viewer output. Should be ready in a couple of hours, but will come with another PR with some viewer changes.

Ok, nice! I'm afraid we can only show GIFs on this README tho...

**Comment by gipert** [comment]
Once this PR is closed, we should also copy paste to the homepage at https://pyg4ometry.readthedocs.io

**Comment by stewartboogert** [comment]
> Thanks Manuel :) I would propose to show this (Orb instead of Box), looks quite pretty:
> @stewartboogert could you record a GIF, as in the pyg4ometry tutorials?

Ok I'll record a gif. Although doing an embedded GLTF view was exactly to avoid the gifs ;-)

**Comment by gipert** [comment]
> So the embedding of course will work in the docs but not the MD readme.

I guess then on the webpage it will show up as a widget responsive to user input? Nice!

**Comment by stewartboogert** [comment]
> > Thanks Manuel :) I would propose to show this (Orb instead of Box), looks quite pretty:
> > @stewartboogert could you record a GIF, as in the pyg4ometry tutorials?
> 
> Ok I'll record a gif. Although doing an embedded GLTF view was exactly to avoid the gifs ;-)
> 
> So the embedding of course will work in the docs but not the MD readme.

Ok so I've made a video, but need to find somewhere nice to locate it. Don't want to pollute the repo and I don't like where the pg4ometry manual videos are places. So I need to come up with a better solution. I'm merge this PR and make an issue about the video. 

All happy with this?

**Pull request merged**